test/test.py

```diff
@@ -19,7 +19,9 @@ def environ(name, default = None):
 	return os.environ[name]
 
 STRICT_LEVEL = int(environ('STRICT_LEVEL', '2'))
+SPEC_LEVEL = int(environ('SPEC_LEVEL', '2'))
```

Comment:
@kalyanaj @basti1302 

Should we test level 2 by default or leave it to testers to manually bump to level 2 testing? My fear is that automated test suites for things like OTel which pull the test suite from git might fail for existing level 1 implementations.

Reply:
Good question. I don't think there is an ideal choice here.

How about this:
* We default to level 1 for now, implementors have to explicitly set the spec level to 2 if they already want to run level 2 tests.
* When level 2 has reached the candidate recommendation stage we set the default level to 2.
    * Yes, this will break the existing test suites of level 1 implementations (assuming they pull the test suite from git via the main branch instead of pinning a specific commit). I think at that point in time, this is actually a good signal for those folks to make them aware that there _is_ a level 2 of the spec.
    * The fix is easy: Add the env var SPEC_LEVEL=1 until you update the implementation to level 2 compliance. We need to point out that env var  very clearly but I think the readme and the help output of the test suite already do a pretty good job. I we want to go the extra mile we could log something at the start of the the test suite when the env var is absent. Something like:
    
```
if SPEC_LEVEL is unset:
    print('Warning: The environment variable SPEC_LEVEL has not been set. Defaulting to specification level {}. Consider setting SPEC_LEVEL explicitly if your implementation is based on a different specification level.'.format(DEFAULT_SPEC_LEVEL))
```

Reply:
Changed to level 1 default and added the suggested warning. I'm still concerned that users that already have it running in CI won't see it but at least the fix is easy in the future.